tests/test_bootstrapping.py

```diff
@@ -1,10 +1,7 @@
 from __future__ import annotations
 
-from typing import Callable
-
```

Comment:
This and many changes below, I don't get what this has to do with formatting. Are those unrelated changes you add in the same PR?

Reply:
These are a few changes that were raised by Ruff (unused imports, naming, etc.)

Reply:
Ok, not sure about this one. Say you are working on a PR and you comment the only line that depends on an import. Then, later on, if you commit, then you want to uncomment the line, you would need to rewrite the import that was formatted out? Or will this take into consideration commented lines too?

Reply:
I'd be ready to give my go, just wanted to know what you think of this comment before

Reply:
We now have much more enforced checks for unused imports. Essentially, if you have an unused import outside a `__init__.py` file, the checker will complain until you either remove it or add a `# noqa: F401` to the import line.